### DISCUSSION TITLE
How can I customize the installation directory of portable applications (Putty, WinSCP, etc)?

### DISCUSSION THREAD

**Discussion by TheQuantumPhysicist** [chocolatey/choco#3378] [Q&A]
I saw that every specific application has a different way for customizing the installation directory, which is OK. How can I customize the installation directories of Putty and WinSCP? Do all portable programs share the same method for doing this?

**Comment by AdmiringWorm** [comment]
Most portable packages available on the Community Repository right now install themselves to Chocolatey's lib package directory (I know Putty does this, but haven't looked at the WinSCP package).

Usually, for packages that do this, you cannot change the installation directory, unfortunately.

There are, however, some packages that use different locations as well (Like installing to a non-admin tools location), this location can be changed using the `ChocolateyToolsLocation` environment variable, but there aren't that many of them.

If a portable package has not documented a way to override the installation directory in the package description, it is very unlikely that you can change the location. You can always reach out to the maintainer of the package (Either through the `Contaict Maintainers` link, or through their `Package Source` link) and ask if they are willing to provide a way to change the installation directory.
